timeskip.lua

```diff
@@ -0,0 +1,146 @@
+-- modify the duration of the "Updating World" timeskip
+
+local usage = [====[
+
+timeskip
+========
+Starting a new fortress/adventurer session is preceded by
+an "Updating World" timeskip which is normally 2 weeks long.
+This script allows you to modify the duration of this timeskip,
+enabling you to jump into the game earlier or later than usual.
+
+You can use this at any point before the timeskip begins
+(for example, while still at the "Start Playing" menu).
+
+It is also possible to run the script during the timeskip,
+which can be useful if you decide to end the process earlier
+or later than initially planned.
+
+Usage::
+
+    -ticks X
+        Replace "X" with the desired timeskip duration in ticks
+        The following conversions may help you calculate this:
+            1 tick = 72 seconds = 1 minute 12 seconds
+            50 ticks = 60 minutes = 1 hour
+            1200 ticks = 24 hours = 1 day
+            8400 ticks = 7 days = 1 week
+            33600 ticks = 4 weeks = 1 month
+            403200 ticks = 12 months = 1 year
+
+    -clear
+        If the timeskip has not yet begun, this resets its duration
+        to the default value.
+
+Example::
+
+    timeskip -ticks 851249
```

Comment:
wouldn't it be more natural for the user to input the desired time skip in calendar time instead of having to convert to ticks? Perhaps an API more like [`dfhack.timeout()`](https://docs.dfhack.org/en/latest/docs/Lua%20API.html#core-interpreter-context) would be better, like:

`timeskip 1 minute`
`timeskip 42 years`
`timeskip 2 years 1 month 9 days 8 hours 58 minutes 48 seconds`

generalized: `timeskip [<years> year[s]] [<months> month[s]] [<days> day[s]] [<minutes> minute[s]] [<seconds> second[s]]`

and then calculate the ticks (rounded up) from that?

Reply:
I considered this whilst planning the script, but decided against it for the following reasons:

- Since 1 tick equates to 72 seconds of calendar time, allowing players to specify `seconds` and `minutes` would probably be misleading and confusing, as the majority of inputs would need to be heavily rounded to conform to this. As such, it seemed to me that specifying ticks directly is more intuitive and precise in this regard (in addition to being simpler to code).

- Calculating tick conversions is quite simple, especially since most users are likely to opt for simple durations along the lines of 5 years, unlike the convoluted value given in the example (which is really only there to better illustrate how ticks translate to calendar time). Of note, players familiar with modding should already have some understanding of this, as things like syndrome durations require time to be specified in ticks. With this in mind, I assumed that the suggested functionality probably wouldn't be worth the added complexity.

That said, I do understand that manual conversions can be a bit of a hassle. Perhaps as a middle ground we could make it possible to specify time in either `ticks` or the more convenient `years`, `months`, `days` and `hours`, omitting `seconds` and `minutes` under the expectation that players should use the `ticks` argument if they want further precision?

Reply:
supporting `-years`, `-months`, `-days`, and `-hours` in addition to `-ticks` sounds perfectly reasonable to me.